hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java

```diff
@@ -168,4 +173,28 @@ public String retryQuestion(String questionId, String comments, Question questio
                 .build());
         return askQuestion(question);
     }
+
+    public BinaryResource getAvatar(String agentId)
```

Comment:
It would be worth caching this methods results as they are quite static and shouldn't change much. 
That doesn't have to come with this PR as I think getting avatars won't be very frequently called but still might be worth it.
Caching might also come handy when we want to reduce temporary file creation needs.

Reply:
I think it's a good idea. I'll prepare a separate PR with this.